website/content/docs/discovery/dns.mdx

```diff
@@ -444,33 +473,46 @@ If you need more complex behavior, please use the
 
 ### Service Virtual IP Lookups
 
-To find the unique virtual IP allocated for a service:
+To find the unique virtual IP allocated for a [Connect-capable](/docs/connect) service:
 
 ```text
 <service>.virtual[.<peer>].<domain>
 ```
 
-This will return the unique virtual IP for any [Connect-capable](/docs/connect)
-service. Each Connect service has a virtual IP assigned to it by Consul - this is used
-by sidecar proxies for the [Transparent Proxy](/docs/connect/transparent-proxy) feature.
-The peer name is an optional part of the FQDN, and it is used to query for the virtual IP
-of a service imported from that peer.
+Consul assigns a virtual IP to each service in the service mesh.
+A downstream service in the mesh must address an upstream service
+using its virtual IP in the following scenarios:
+- The upstream service is imported from a cluster peer
```

Comment:
You can still address an upstream service imported from a cluster peer using explicit upstreams. In that case you wouldn't need to use the virtual IP. So the only time we must address an upstream using its virtual ip is when the downstream has tproxy enabled. 

"The peer name is an optional part of the FQDN, and it is used to query for the virtual IP
of a service imported from that peer." <- i think this sentence is still accurate

Reply:
@ndhanushkodi : So if using explicit upstreams, you can use either a `.virtual` or a `.service` lookup? I guess I'm still confused about when you'd need to use a `.virtual` lookup versus just using a `.service` lookup all the time (as of 1.14.2).

Reply:
Is the only time we need `.virtual` "when the downstream has tproxy enabled"? And in all other cases, use `.service`?

Reply:
@jkirschner-hashicorp If a downstream application is using an explicit upstream, it would be communicating to `localhost:<port>`, and would not be using the IP address returned from a Consul DNS lookup.

Accessing a virtual address _requires_ the use of tproxy. The VIPs are allocated from an IP range that is not likely to be present on a user's internal network, and thus are not routable by the upstream network. The application's sidecar needs to have tproxy enabled so that Envoy is correctly configured to map the destination [V]IP to the correct upstream service.

Reply:
> If a downstream application is using an explicit upstream, it would be communicating to localhost:<port>, and would not be using the IP address returned from a Consul DNS lookup.

@blake : I don't think I had ever considered how namespace, partition, and peer are specified in an explicit upstream. Somehow I had thought that format would use a DNS-style lookup, but it doesn't. It uses `proxy.upstreams[].destination_*` fields.

I think I understand now:
- Implicit upstreams with transparent proxy must use the `.virtual` format (if using Consul DNS format rather than Kube DNS format)
- DNS formats doesn't come into the picture at all for explicit upstreams
- For service discovery DNS lookup, use `.service`.

---

Side note-to-self: We don't document `destination_peer` (if there is such a field) in the upstream config reference: https://developer.hashicorp.com/consul/docs/connect/registration/service-registration#upstream-configuration-reference

It also seems like there's duplicate content between https://developer.hashicorp.com/consul/docs/connect/registration/sidecar-service and the link above that can lead to inconsistencies, like the omission of destination namespace and partition in one place.

Reply:
Though, use of this `.virtual` format requires that [`dns.enableRedirection` is set to `true`](https://developer.hashicorp.com/consul/docs/k8s/helm#v-dns-enableredirection), right?

Reply:
> We don't document destination_peer (if there is such a field) in the upstream config reference: https://developer.hashicorp.com/consul/docs/connect/registration/service-registration#upstream-configuration-reference

This field *is* documented on that page. At the time of writing this comment, it is documented in the third row in the table of supported parameters under [Upstream configuration reference](https://developer.hashicorp.com/consul/docs/connect/registration/service-registration#upstream-configuration-reference).

> It also seems like there's duplicate content between https://developer.hashicorp.com/consul/docs/connect/registration/sidecar-service…

This page shows the [default config options](https://developer.hashicorp.com/consul/docs/connect/registration/sidecar-service#sidecar-service-defaults) used by Consul when a service registration has Connect enabled using the minimal `"connect": { "sidecar_service": {} }` config stanza. It also shows an example of how these parameters can be overridden with minimal additional configuration.

The [/docs/connect/registration/service-registration](https://developer.hashicorp.com/consul/docs/connect/registration/service-registration) contains an exhaustive list of all parameters supported for a proxy service registration.

Reply:
> This field is documented on that page. At the time of writing this comment, it is documented in the third row in the table of supported parameters under [Upstream configuration reference](https://developer.hashicorp.com/consul/docs/connect/registration/service-registration#upstream-configuration-reference).

Ugh, didn't see that. Thanks!